📖  Add steps to resolve Docker Desktop inotify limit issues (#5947)
kubernetes-sigs/cluster-api: docs/book/src/user/troubleshooting.md

    * Run [docker system prune --volumes](https://docs.docker.com/engine/reference/commandline/system_prune/) to prune dangling images, containers, volumes and networks.

## Cluster API with Docker Desktop - "too many open files"

> When using Cluster API and Docker infrastructure on MacOS with Docker Desktop an error of "too many open files" has been observed.

Reviewer: ```suggestion
When using Cluster API and the Docker infrastructure provider on MacOS with Docker Desktop an error of "too many open files" has been observed.
```
nit

One solution to this issue is to increase the maximum inotify file watch settings in the Docker Desktop VM.

To do so:

1) Enter the Docker Desktop VM
```
nc -U ~/Library/Containers/com.docker.docker/Data/debug-shell.sock
```
2) Increase the inotify limits using sysctl
```
sysctl fs.inotify.max_user_watches=1048576
sysctl fs.inotify.max_user_instances=8192
```
3) Exit the Docker Desktop VM
```
exit
```

> Note: This error was observed in the upgrade to Docker 4.3. An alternative solution is to stick with another version of Docker while this issue is being resolved.

Reviewer: ```suggestion
Note: This error was observed with Docker 4.3. An alternative solution is to stick with an older version of Docker while this issue is being resolved.
```

Let's also link the upstream issue: https://github.com/docker/for-mac/issues/6071
